python/vowpalwabbit/pyvw.py

```diff
@@ -114,7 +114,12 @@ def format_input(key, val):
     def parse(self, str_ex, labelType=pylibvw.vw.lDefault):
         """Returns a collection of examples for a multiline example learner or a single
         example for a single example learner."""
+        if not (isinstance(str_ex, list) or isinstance(str_ex, str)):
+            raise TypeError('Unsupported type. List or string object must be passed.')
+        if isinstance(str_ex, list):
+            str_ex = "\n".join(str_ex)
         str_ex = str_ex.replace('\r', '')
+        str_ex = str_ex.strip()
```

Comment:
Whats the reasoning behind needing strip?

Reply:
I have used it because while running the tests and observing the error, I found that the error was due to string ending with '\n' as in 3rd case. So I used strip to remove trailing spaces and '\n'. I think it does effect the results of other formats because it only removes trailing spaces in a string.

Reply:
I gone through the tutorial of `--cd explore_adf` which stated that each of the example starts with as new line. Here if the string ends with '\n' , then it would consider that another example has been encountered and number of examples in 3rd case will rise to 3.

Reply:
Yep that makes sense, thanks!